src/app/Http/Controllers/V1/DocumentDistributionController.php

```diff
@@ -0,0 +1,347 @@
+        $inbox->NId                 = $this->inboxId;
+        $inbox->CreatedBy           = $this->creator?->PeopleId;
+        $inbox->CreationRoleId      = $this->creator?->PrimaryRoleId;
+        $inbox->NTglReg             = Carbon::now();
+        $inbox->Tgl                 = Carbon::parse($this->document->tanggal)->format('Y-m-d H:i:s');
+        $inbox->JenisId             = $this->findDocumentTypeId();
+        $inbox->UrgensiId           = $this->findDocumentUrgency();
+        $inbox->SifatId             = $this->findDocumentConfidentiality();
+        $inbox->Nomor               = $this->document->nomor_akhir;
+        $inbox->Hal                 = $this->document->perihal;
+        $inbox->Pengirim            = 'internal';
+        $inbox->NTipe               = $this->findDocumentDraftType();
+        $inbox->Namapengirim        = $this->creator?->role->RoleDesc;
+        $inbox->NFileDir            = 'naskah';
+        $inbox->BerkasId            = '1';
+        $inbox->ClId                = $this->findClassificationId();
+        $inbox->sidebar_v2_tid      = $this->document->id;
+        $inbox->sidebar_v2_metadata = json_encode($this->document);
+        $inbox->tandatangan         = json_encode($this->document->tandatangan);
+        $inbox->save();
+    }
+
+    /**
+     * saveToInboxReceiver
+     * Save document to inbox_receiver table based on payload document v2
+     *
+     * @return void
+     */
+    protected function saveToInboxReceiver()
+    {
+        $InboxReceiver = new InboxReceiver();
+        $InboxReceiver->NId           = $this->inboxId;
+        $InboxReceiver->NKey          = TableSetting::first()->tb_key;
+        $InboxReceiver->GIR_Id        = $this->inboxId;
+        $InboxReceiver->From_Id       = $this->creator?->PeopleId;
+        $InboxReceiver->RoleId_From   = $this->creator?->PrimaryRoleId;
+        $InboxReceiver->To_Id         = $this->receiver?->PeopleId;
+        $InboxReceiver->RoleId_To     = $this->receiver?->PrimaryRoleId;
+        $InboxReceiver->ReceiverAs    = 'to_forward';
+        $InboxReceiver->StatusReceive = 'unread';
+        $InboxReceiver->ReceiveDate   = Carbon::now();
+        $InboxReceiver->To_Id_Desc    = $this->receiver?->role->RoleDesc;
+        $InboxReceiver->Status        = '0';
+        $InboxReceiver->action_label  = ActionLabelTypeEnum::REVIEW();
+        $InboxReceiver->save();
+    }
+
+    /**
+     * doSaveToInboxFile
+     * Transfer file to webhook v1 and save inbox files
+     *
+     * @return void
+     */
+    protected function doSaveToInboxFile()
+    {
+        // Setup pdf file name
+        $type = str_replace(' ', '_', ucwords($this->document->jenis));
+        $title = str_replace(' ', '_', trim(preg_replace('/[^a-zA-Z0-9_ -]/s', '', substr($this->document->perihal, 0, 180))));
+        $time = parseDateTimeFormat(Carbon::now(), 'dmY') . '_' . parseDateTimeFormat(Carbon::now(), 'His');
```

Comment:
rather than this, isnt it easier to just use `format()` method from carbon?

```suggestion
        $time = Carbon::now()->format('Ymd_His');
```

Reply:
if we didnt use the parse date time format, it will be on gmt format

Reply:
i see. how about this then?

```suggestion
        $time = parseDateTimeFormat(Carbon::now(), 'dmY_His');
```

Reply:
naaah i now what you mean mas. thanks for suggestion

Reply:
fixed here [a6bd0b0](https://github.com/jabardigitalservice/office-services/pull/660/commits/a6bd0b0b0e6610f5b0d4cb38b9431284444443d2)